File: C++/405. Convert a Number to Hexadecimal.cpp
class Solution {
public:
    string toHex(int num) {
        string a = "0123456789abcdef";
        if(num==0) return "0";
        string ans = "";
        int cnt = 0;
        while(num && cnt<8)
        {
            ans += a[(num & 0xf)];
            num >>= 4;
            cnt++;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: Python/405. Convert a Number to Hexadecimal.py
class Solution:
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        h = '0123456789abcdef'
        if num == 0:
            return '0'
        ans, cnt = '', 0
        while num and cnt < 8:
            ans += h[num & 0xf]
            num >>= 4
            cnt += 1
        return ans[::-1]